docs: improve readme (#2518)
containerbase/buildpack: README.md

[![codecov](https://codecov.io/gh/containerbase/base/branch/main/graph/badge.svg?token=GYS2ZZAXDP)](https://codecov.io/gh/containerbase/base)

This repository is the source for the Docker images [`containerbase/base`](https://hub.docker.com/r/containerbase/base) and `ghcr.io/containerbase/base`.
The commits to the `main` branch are automatically built and published.

## Local development

> You need a recent [Docker](https://www.docker.com) version and install the latest [`buildx`](https://github.com/docker/buildx) plugin.

Reviewer: maybe just `you need a recent docker and buildx version`?

Author: ```suggestion
Install a recent version of:

- [Docker](https://www.docker.com)
- the [`buildx`](https://github.com/docker/buildx) plugin
```

> You need a recent [Docker](https://www.docker.com) version and install the [`buildx`](https://github.com/docker/buildx) [`>= v0.4.0`](https://github.com/docker/buildx/releases/tag/v0.4.0) plugin.

Reviewer: ```suggestion
You need a recent [Docker](https://www.docker.com) version and install the latest [`buildx`](https://github.com/docker/buildx) plugin.
```
we should link to the latest release. maybe we already need a higher version. 🤔

let's ensure latest version

### Base image

> If you make changes to the [`src`](./src/) folder or the [`Dockerfile`](./Dockerfile), you must rebuild the `containerbase/base` image.

Reviewer: pnpm build needs also be called

Author: ```suggestion
If you make changes to the [`src`](./src/) folder or the [`Dockerfile`](./Dockerfile), you must:

- rebuild the `containerbase/base` image
- run `pnpm build`
```

Reviewer: pnpm build first, then image build

Author: I made an ordered list and switched the order of the commands. 😉

### Base image

If you make changes to the [`src`](./src/) folder or the [`Dockerfile`](./Dockerfile), you must:

1. run `pnpm build`
1. rebuild the `containerbase/base` image

```sh

> docker buildx bake

Reviewer: ```suggestion
pnpm build
docker buildx bake
```
maybe add both commands here?

```

You can use the following command to ignore the remote cache for local testing.
This may speed up your local builds.

```sh
docker buildx bake  --set *.cache-from=
```

### Test images

To run one of the tests use the following command, it will run the Java tests from [`test/java`](./test/java/).

```sh

> TAG=java docker buildx bake test

Reviewer: I'll make new suggestions tomorrow, there's a simpler pnpm command now

Reviewer: You can run `pnpm test:docker -b -t java` instead. the `-b` causes the required binaries to rebuild before testing.

`pnpm test:docker` causes all language tests to run.

Author: Can you suggest the changes you want?

Reviewer: let's leave it as is for now. it's not wrong.

### Base image

If you make changes to the [`src`](./src/) folder or the [`Dockerfile`](./Dockerfile), you must:

1. run `pnpm build`
1. rebuild the `containerbase/base` image

```sh
pnpm build
docker buildx bake
```

You can use the following command to ignore the remote cache for local testing.
This may speed up your local builds.

```sh
docker buildx bake  --set *.cache-from=
```

### Test images

To run one of the tests use the following command, it will run the Java tests from [`test/java`](./test/java/).

```sh
TAG=java docker buildx bake test
```

For other test images see the [`test`](./test/) folder.

### Distro test images

> To run the `jammy` tests use the following command, it will run the test from [`test/Dockerfile.jammy`](./test/Dockerfile.jammy).

Author: Is there a `noble` test for Ubuntu `24.04`, the newest LTS?

Reviewer: yes, added recently

> To run the `noble` tests use the following command, it will run the test from [`test/Dockerfile.noble`](./test/Dockerfile.noble).

Reviewer: 🤔 `jammy` is still a valid target.

Author: What do you want?

- both `jammy` and `noble`
- only `jammy`
- only `noble`
- something else

Reviewer: both

```sh

> TAG=jammy docker buildx bake test-distro

Author: Also, is there a `TAG=noble`?

Reviewer: yes
